pkg/actors/actor.go

```diff
@@ -7,12 +7,60 @@ package actors
 
 import (
 	"sync"
+	"sync/atomic"
 	"time"
+
+	diag "github.com/dapr/dapr/pkg/diagnostics"
 )
 
 type actor struct {
-	lock         *sync.RWMutex
-	lastUsedTime time.Time
-	busy         bool
-	busyCh       chan (bool)
+	actorType string
+	actorID   string
+
+	concurrencyLock *sync.RWMutex
+	lastUsedTime    time.Time
+	busy            bool
+	busyCh          chan (bool)
+
+	pendingLocks int32
+}
+
+func newActor(actorType, actorID string) *actor {
+	return &actor{
+		actorType:       actorType,
+		actorID:         actorID,
+		concurrencyLock: &sync.RWMutex{},
+		busy:            true,
+		busyCh:          make(chan bool, 1),
+		lastUsedTime:    time.Now().UTC(),
+	}
+}
+
+func (a *actor) isBusy() bool {
+	return a.busy
+}
+
+func (a *actor) channel() chan (bool) {
+	return a.busyCh
+}
+
+func (a *actor) lock() {
+	pLocks := atomic.AddInt32(&a.pendingLocks, 1)
```

Comment:
Nit: this name sounds a bit like hungarian notation.  I'd prefer `pendingLocks`.

Reply:
I removed pLocks and just use member field.